src/AzureClient/EntryPoint/EntryPoint.cs

```diff
@@ -73,12 +84,78 @@ public Task<IQuantumMachineJob> SubmitAsync(IQuantumMachine machine, AzureSubmis
                 }
             }
 
-            var entryPointInput = parameterValues.Count switch
+            return parameterValues.Count switch
             {
                 0 => QVoid.Instance,
                 1 => parameterValues.Single(),
                 _ => InputType.GetConstructor(parameterTypes.ToArray()).Invoke(parameterValues.ToArray())
             };
+        }
+
+        private ArgumentValue GetArgumentValue(string parameterValue, System.Reflection.ParameterInfo parameter)
+        {
+            var parameterType = parameter.ParameterType;
+
+            if (parameterType == typeof(bool))
+            {
+                return new ArgumentValue.Bool(Newtonsoft.Json.JsonConvert.DeserializeObject<bool>(parameterValue));
+            }
+            else if (parameterType == typeof(double))
```

Comment:
(nit)
In the body of this function (`GetArgumentValue()`) we always access the second parameter named `parameter`, and we either access or don't access (throw an exception) the first parameter named `parameterValue`. From this point of view, strictly speaking, the first parameter named `parameterValue` is _optional_. 
I would recommend to consider changing places of the parameters named `parameter` and `parameterValue`.

Reply:
done